### PR TITLE
microk8s

### PR DESCRIPTION
**Pull request by hasheddan**
Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

### Description of your changes
Added a script to make developing Crossplane locally using [MicroK8s](https://microk8s.io/) a much more efficient endeavor. Follows similar patterns to the [minikube.sh](https://github.com/crossplaneio/crossplane/blob/master/cluster/local/minikube.sh) and [local.sh](https://github.com/crossplaneio/crossplane/blob/master/cluster/local/local.sh) scripts. Most significant difference is the usage of MicroK8s internal registry which can be confusing for users not familiar with containerd.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: cluster/local/README.md [hunk 1]
```diff
@@ -42,8 +44,34 @@ For complete list of subcommands supported by `minikube.sh`, run:
 cluster/local/minikube.sh
 ```
 
+#### MicroK8s (recommended for Linux)
+Starting the cluster on Minikube is as simple as running:
```
Comment (displague) [inline cluster/local/README.md]: `Minikube` reference.

File: cluster/local/README.md [hunk 2]
```diff
@@ -42,8 +44,34 @@ For complete list of subcommands supported by `minikube.sh`, run:
...
+```
+
+For complete list of subcommands supported by `microk8s.sh`, run:
+```console
+cluster/local/microk8s.sh
+```
+
 ## Run Tests
-From the project root do the following:
+From the project root do the following (minikube used for example, but same pattern can be followed with any Kubernetes distribution):
```
Comment (displague) [inline cluster/local/README.md]: I would reword this.

```
The following sections provide commands helpful for development environments.  
`minikube.sh` may be replaced with [`local.sh`](link-to-above) or [`microk8s.sh`](link-to-above) in those environments.
These commands expect to be run from the project root.
```

Comment (displague) [reply cluster/local/README.md]: I don't know that these scripts are actually dependent on being run from the root.

Comment (hasheddan) [reply cluster/local/README.md]: I like the rewording. The reason it specifies root is because of the `make` commands and the full paths for the scripts.

File: cluster/local/README.md (hunk 1 above)
Comment (hasheddan) [reply cluster/local/README.md]: Good catch!

File: cluster/local/README.md [hunk 3]
```diff
@@ -42,8 +44,34 @@ For complete list of subcommands supported by `minikube.sh`, run:
...
+```
+
+Resetting the MicroK8s cluster can be done with:
+```console
+cluster/local/microk8s.sh clean
+```
+
+Stopping the MicroK8s cluster can be done with:
+```console
+cluster/local/microk8s.sh clean
```
Comment (suskin) [inline cluster/local/README.md]: Should this be the `stop` command?

Comment (hasheddan) [reply cluster/local/README.md]: Yes! Good catch.

File: cluster/local/microk8s.sh [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+# Deletes pods with application prefix. Namespace is expected as the first argument
+function delete_pods() {
+    for pod in $(kubectl get pods -n "$2" -l "app=$1" --no-headers -o custom-columns=NAME:.metadata.name); do
+        kubectl delete pod "$pod" -n "$2"
+    done
+}
+
+# current kubectl context == minikube, returns boolean
+function check_context() {
+    if [ "$(kubectl config view 2>/dev/null | awk '/current-context/ {print $NF}')" = "minikube" ]; then
```
Comment (displague) [inline cluster/local/microk8s.sh]: This is returning true (`0`) when "minikube" is the current context.
I would assume this should be `"microk8s"` so that the "restart" command works correctly.

In my local configuration, based on this branch, `kubectl config get-contexts` and `kubectl config get-clusters` doesn't show a "microk8s" value.. The `awk` expression above results in `""` for me.

File: cluster/local/microk8s.sh [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    microk8s.start
+
+    microk8s.enable ingress
+
+    microk8s.enable registry
+
+    kubectl apply -f ${scriptdir}/helm-rbac.yaml
+    ${HELM} init --service-account tiller
+
+    copy_image_to_cluster ${BUILD_IMAGE} ${MICROK8S_IMAGE}
```
Comment (displague) [inline cluster/local/microk8s.sh]: When I first ran `microk8s.sh up`, port `32000` was not yet ready.  It took a minute or so to become available,

**Comment by hasheddan** [comment]
I believe this PR has received sign-off from reviewers. Does further action need to be taken to get it merged? @negz @jbw976
